whylog/rulesbase/__init__.py

```diff
@@ -1,6 +1,144 @@
+import re
+import yaml
+from abc import ABCMeta, abstractmethod
+from collections import defaultdict
 
+class AbstractConfigAccess(object):
+    __metaclass__ = ABCMeta
 
-class WhylogBase(object):
+    @abstractmethod
+    def get_parsers(self):
+        pass
 
-    def __init__(self, **kwargs):
+    @abstractmethod
+    def get_rule(self, effect_name):
         pass
+        
+class YamlConfigAccess(AbstractConfigAccess):
+    def __init__(self, parsers_path, rules_path):
+        self._parsers_path = parsers_path
+        self._rules_path = rules_path
+        self._load_parsers()
+        self._load_rules()
+
+    def _load_config_from_file(self, path):
+        with open(path, "r") as config_file:
+            return list(yaml.load_all(config_file))
+
+    def _get_params_for_parser(self, parser_dict):
+        if 'params' in parser_dict:
+            return parser_dict.get('params')
+        else:
+            return None
+
+    def _load_parsers(self):
+        parsers = self._load_config_from_file(self._parsers_path)
+        parsers_dict = {}
+        for parser in parsers:
+            parsers_dict[parser.get('name')] = Parser(
+                parser.get('name'), 
+                parser.get('regex'), 
+                self._get_params_for_parser(parser)
+            )
+        self._parsers = parsers_dict
+
+    def _get_causes_for_rule(self, rule_dict):
```

Comment:
this method suggests that an actual config can have cause (one) or causes (many). This is probably a design error - in all instances there should be multiple causes.